src/e2e-node/e2e.test.ts

```diff
@@ -536,5 +538,159 @@ describe.each(serversUnderTest)(
       await saveAclFor(datasetWithAcl, cleanedAcl, { fetch: session.fetch });
       await deleteSolidDataset(datasetWithoutAclUrl, { fetch: session.fetch });
     });
+
+    describe("Access Control Policies", () => {
+      if (
+        rootContainer.includes("inrupt.net") ||
+        rootContainer.includes("pod-compat.inrupt.com")
+      ) {
+        // These servers do not support Access Control Policies,
+        // so ACP tests can be skipped for them:
+        return;
+      }
+
+      async function initialisePolicyResource(
+        policyResourceUrl: UrlString,
+        session: Session
+      ) {
+        let publicRule = acp.createRule(policyResourceUrl + "#rule-public");
+        publicRule = acp.setPublic(publicRule, true);
+
+        let publicReadPolicy = acp.createPolicy(
+          policyResourceUrl + "#policy-publicRead"
+        );
+        publicReadPolicy = acp.addRequiredRuleUrl(publicReadPolicy, publicRule);
```

Comment:
would it make sense if the rule containing the public agent was an optional one (`acp:any`) instead of a required one (`acp:all`) ? I think in this specific context it would not change anything, so I don't think it's necessary to change the current implementation, it's mostly a question to check that I understand what's going on ^^

Reply:
Yeah that's one problem I have with the current API: I called it `Optional`, but it's not optional at all - at least one of those rules should be valid.

It would be good though if both variants would be checked, so I'll change one of the required rules to be "optional".

Reply:
7e2b9f5ffae53f5ba16e11aba6852f0fb6aec3ec